edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/sdkClient/WrappingSdkModuleClient.cs

```diff
@@ -14,7 +15,18 @@ public class WrappingSdkModuleClient : ISdkModuleClient
         public WrappingSdkModuleClient(ModuleClient sdkModuleClient)
             => this.sdkModuleClient = Preconditions.CheckNotNull(sdkModuleClient, nameof(sdkModuleClient));
 
-        public Task OpenAsync() => this.sdkModuleClient.OpenAsync();
+        public Task OpenAsync()
+        {
+            try
+            {
+                return this.sdkModuleClient.OpenAsync();
+            }
+            catch (Exception)
+            {
+                this.sdkModuleClient?.Dispose();
```

Comment:
do we need to close first before calling dispose?

Reply:
no need to close, confirmed with SDK team that we should use Dispose instead as is cleaning everything and Close has some gaps.